airflow/models/taskinstance.py

```diff
@@ -1207,9 +1252,15 @@ def get_dagrun(self, session: Session = NEW_SESSION) -> DagRun:
 
         return dr
 
+    @staticmethod
+    @internal_api_call
     @provide_session
     def check_and_change_state_before_execution(
-        self,
+        dag_id: str,
```

Comment:
Primary key for task instance is composite with `dag_id`, `run_id`, `task_id`, `map_index`. So the call is a little bit more verbose